File: ios_app/iphone/UI/Home/HomeChecklistHintView.swift
import SwiftUI
import shared

struct HomeChecklistHintView: View {
    
    let hintUi: HomeVm.ChecklistHintUi
    
    ///
    
    @Environment(Navigation.self) private var navigation
    
    var body: some View {
        HStack {
            
            Button(
                action: {
                    hintUi.create(
                        dialogsManager: navigation,
                        onSuccess: { checklistDb in
                            navigation.sheet {
                                ChecklistFormItemsSheet(
                                    checklistDb: checklistDb,
                                    onDelete: {},
                                )
                            }
                        },
                    )
                },
                label: {
                    HStack {
                        
                        ZStack {
                            Image(systemName: "plus")
                                .foregroundColor(.black)
                                .font(.system(size: 13, weight: .semibold))
                        }
                        .frame(width: HomeScreen__itemCircleHeight, height: HomeScreen__itemCircleHeight)
                        .background(Circle().fill(.white))
                        .padding(.leading, HomeScreen__hPadding)
                        .padding(.trailing, HomeScreen__itemCircleMarginTrailing)
                        
                        Text(hintUi.title)
                            .foregroundColor(.white)
                            .font(.system(size: HomeScreen__primaryFontSize))
                            .frame(height: HomeScreen__itemHeight)
                    }
                }
            )
            
            Spacer()
            
            Button(
                action: {
                    hintUi.hide()
                },
                label: {
                    ZStack {
                        Image(systemName: "xmark")
                            .foregroundColor(.black)
                            .font(.system(size: 13, weight: .semibold))
                    }
                    .frame(width: HomeScreen__itemCircleHeight, height: HomeScreen__itemCircleHeight)
                    .background(Circle().fill(Color(.systemGray2)))
                    .padding(.trailing, HomeScreen__hPadding)
                }
            )
        }
    }
}

File: android/src/main/java/me/timeto/app/ui/home/HomeChecklistHintView.kt
package me.timeto.app.ui.home

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.Icon
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import me.timeto.app.R
import me.timeto.app.ui.HStack
import me.timeto.app.ui.SpacerW1
import me.timeto.app.ui.ZStack
import me.timeto.app.ui.c
import me.timeto.app.ui.checklists.form.ChecklistFormItemsFs
import me.timeto.app.ui.navigation.LocalNavigationFs
import me.timeto.app.ui.roundedShape
import me.timeto.shared.vm.home.HomeVm

@Composable
fun HomeChecklistHintView(
    hintUi: HomeVm.ChecklistHintUi,
) {

    val navigation = LocalNavigationFs.current

    HStack {

        HStack(
            modifier = Modifier
                .clickable {
                    hintUi.create(
                        dialogsManager = navigation,
                        onSuccess = { checklistDb ->
                            navigation.push {
                                ChecklistFormItemsFs(
                                    checklistDb = checklistDb,
                                    onDelete = {},
                                )
                            }
                        },
                    )
                }
        ) {

            ZStack(
                modifier = Modifier
                    .padding(
                        start = HomeScreen__hPadding,
                        end = HomeScreen__itemCircleMarginTrailing,
                    )
                    .size(HomeScreen__itemCircleHeight)
                    .clip(roundedShape)
                    .background(c.white),
                contentAlignment = Alignment.Center,
            ) {
                Icon(
                    painterResource(id = R.drawable.sf_plus_medium_bold),
                    contentDescription = "Timer Info",
                    tint = c.black,
                    modifier = Modifier
                        .size(10.dp),
                )
            }

            Text(
                text = hintUi.title,
                modifier = Modifier
                    .height(HomeScreen__itemHeight),
                color = c.white,
                fontSize = HomeScreen__primaryFontSize,
            )
        }

        SpacerW1()

        ZStack(
            modifier = Modifier
                .padding(end = HomeScreen__hPadding)
                .size(HomeScreen__itemCircleHeight)
                .clip(roundedShape)
                .background(c.gray2)
                .clickable {
                    hintUi.hide()
                },
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                painterResource(id = R.drawable.sf_xmark_small_medium),
                contentDescription = "Timer Info",
                tint = c.black,
                modifier = Modifier
                    .size(10.dp),
            )
        }
    }
}
